### PR TITLE
Border improvements

### PR DESCRIPTION
**Pull request by MightyPlaza**
fixes issues with window and workspace border related rules not working as intended
also simplifies getting window border size and setting the workspace rules

closes: https://github.com/hyprwm/Hyprland/issues/2957
shouldn't case any issues

### CODE REVIEW

File: src/Window.cpp [hunk 1]
```diff
@@ -589,10 +581,9 @@ void CWindow::updateDynamicRules() {
 // it is assumed that the point is within the real window box (m_vRealPosition, m_vRealSize)
 // otherwise behaviour is undefined
 bool CWindow::isInCurvedCorner(double x, double y) {
-    static auto* const ROUNDING   = &g_pConfigManager->getConfigValuePtr("decoration:rounding")->intValue;
-    static auto* const BORDERSIZE = &g_pConfigManager->getConfigValuePtr("general:border_size")->intValue;
+    static auto* const ROUNDING = &g_pConfigManager->getConfigValuePtr("decoration:rounding")->intValue;
 
-    if (BORDERSIZE >= ROUNDING || ROUNDING == 0)
+    if (getRealBorderSize() >= *ROUNDING || *ROUNDING == 0)
```
Comment (MightyPlaza) [inline src/Window.cpp:586]: not sure if this correct, but doesn't make sense otherwise

Comment (vaxerski) [reply src/Window.cpp:586]: we could switch this over to `CWindow::rounding()` while we're at it. But yeah, looks alright.

**Review by vaxerski — DISMISSED** [review]
the specialrenderdata / additionalconfigdata situation is a mess, it needs to be cleaned up. Note to self...

**Comment by MightyPlaza** [comment]
> the specialrenderdata / additionalconfigdata situation is a mess, it needs to be cleaned up. Note to self...

I though about making a windowProps object that could be set using hyprctl or window rules and workspace rules (maybe combine them into something new to reduce the number of workspace and window rules and the need to update each one every time we want to add something new).
each thing that could set the property would have a certain priority (hyprctl > window rules > workspace rules > layout requests > global config)
but that was more work, and would be storing a lot of redundant -1 most of the time (no idea if there is a better way to do this)

using rounding() now so should be ready for merge @vaxerski

**Comment by vaxerski** [comment]
> I though about making a windowProps object that could be set using hyprctl or window rules and workspace rules (maybe combine them into something new to reduce the number of workspace and window rules and the need to update each one every time we want to add something new).
each thing that could set the property would have a certain priority (hyprctl > window rules > workspace rules > layout requests > global config)
but that was more work, and would be storing a lot of redundant -1 most of the time (no idea if there is a better way to do this)

I will probably look into it when I am back from holidays. There are a bunch of pain points I want to address, but it's way nicer to code on your desktop than on a small laptop in the middle of nowhere.

**Pull request merged**